### DISCUSSION TITLE
code menu option generating <p><code></code></p>

### DISCUSSION THREAD

**Discussion by yogeshbansal** [sibiraj-s/ngx-editor#523] [General]
when the user types a code in the editor its generates a paragraph tag instead of pre tag (see the screenshot below)

So when a user previews the editor content it looks like
<img width="797" alt="Screenshot 2024-03-05 at 12 27 37 PM" src="https://github.com/sibiraj-s/ngx-editor/assets/4295712/12b1f902-957e-4a18-b61f-49a5c8956068">
<img width="542" alt="Screenshot 2024-03-05 at 12 27 44 PM" src="https://github.com/sibiraj-s/ngx-editor/assets/4295712/784d0cf6-96d1-40d5-b855-9a95ff0fb407">
<img width="232" alt="Screenshot 2024-03-05 at 12 37 37 PM" src="https://github.com/sibiraj-s/ngx-editor/assets/4295712/69f7119a-bdbf-49d2-ac56-4456e6bd65e4">

**Comment by sibiraj-s** [comment]
p tag is just the default text wrapping tag for any [markers](https://discuss.prosemirror.net/t/discussion-what-are-marks/862). code here is a marker, hence that way. 

Note: if you are expecting the code to be syntax highlighting version like that, this is not it, this is simply a [code](https://www.w3schools.com/tags/tag_code.asp) tag, for the highlighting you might want to look at https://sibiraj-s.github.io/ngx-editor/examples/codemirror/

**Reply by yogeshbansal** [reply]
got it thanks. 
one more question, for each line of code it creates one pre > code tag, is it possible to have only one pre > code for the whole cod block?
For reference see the image below each line of code is in a separate tag 
<img width="564" alt="Screenshot 2024-03-12 at 5 40 40 AM" src="https://github.com/sibiraj-s/ngx-editor/assets/4295712/31929a37-234e-4599-93d8-af02e9709510">

[a comment by an automated account is not shown]
